File: python/hamming/hamming.py
def distance(nucleotide1, nucleotide2):
    """
    Calculate the Hamming difference between two DNA strands.
    
    :param nucleotide1: first dna strand string
    :param nucleotide2: second dna strand string
    :return: 
    """
    difference_count = 0
    if len(nucleotide1) != len(nucleotide2):
        raise ValueError('DNA strands must be of the same length')
    for nuc1, nuc2 in zip(nucleotide1, nucleotide2):
        if nuc1 != nuc2:
            difference_count += 1
    return difference_count

File: typescript/hamming/hamming.ts
export default class Hamming {

    compute(strand1: string, strand2: string): number {
        let difference_count: number = 0
        const strand1_arr: string[] = strand1.split('')
        const strand2_arr: string[] = strand2.split('')
        if (strand1.length !== strand2.length) {
            throw new Error('DNA strands must be of equal length.')
        }
        strand1_arr.map((val, i) => {
            if (val !== strand2_arr[i]) {
                difference_count++
            }
        })
        return difference_count
    }
}